core/DataArray.php

```diff
@@ -133,11 +127,50 @@ protected function doSumVisitsMetrics($newRowToAdd, &$oldRowToUpdate, $onlyMetri
         $oldRowToUpdate[Metrics::INDEX_NB_VISITS_CONVERTED] += $newRowToAdd[Metrics::INDEX_NB_VISITS_CONVERTED];
     }
 
+    /**
+     * Adds the given row $newRowToAdd to the existing  $oldRowToUpdate passed by reference
+     * The rows are php arrays Name => value
+     *
+     * @param array $newRowToAdd
+     * @param array $oldRowToUpdate
+     * @param bool $onlyMetricsAvailableInActionsTable
+     *
+     * @return void
+     */
+    protected function doSumActionsMetrics($newRowToAdd, &$oldRowToUpdate)
+    {
+        // Pre 1.2 format: string indexed rows are returned from the DB
+        // Left here for Backward compatibility with plugins doing custom SQL queries using these metrics as string
+        if (!isset($newRowToAdd[Metrics::INDEX_NB_VISITS])) {
+            $oldRowToUpdate[Metrics::INDEX_NB_VISITS] += $newRowToAdd['nb_visits'];
+            $oldRowToUpdate[Metrics::INDEX_NB_ACTIONS] += $newRowToAdd['nb_actions'];
+            $oldRowToUpdate[Metrics::INDEX_NB_UNIQ_VISITORS] += $newRowToAdd['nb_uniq_visitors'];
+            return;
+        }
+
+        // Edge case fail safe
+        if (!isset($oldRowToUpdate[Metrics::INDEX_NB_VISITS])) {
+            return;
+        }
+
+        $oldRowToUpdate[Metrics::INDEX_NB_VISITS] += $newRowToAdd[Metrics::INDEX_NB_VISITS];
+        if (array_key_exists(Metrics::INDEX_NB_ACTIONS, $newRowToAdd)) {
+            $oldRowToUpdate[Metrics::INDEX_NB_ACTIONS] += $newRowToAdd[Metrics::INDEX_NB_ACTIONS];
+        }
+        if (array_key_exists(Metrics::INDEX_PAGE_NB_HITS, $newRowToAdd)) {
```

Comment:
We noticed sometimes there are hits and sometimes there are actions, needs to work with both